psm_utils/io/idxml.py

```diff
@@ -389,7 +400,7 @@ def _create_new_ids(self, psm_dict: dict) -> None:
                     # Fill PeptideIdentification object with PeptideHits
                     peptide_id = oms.PeptideIdentification()
                     peptide_id.setMetaValue("spectrum_reference", spectrum_id)
-                    peptide_id.setMetaValue("id_merge_index", msrun_reference.index(run))
+                    peptide_id.setMetaValue("id_merge_index", msrun_reference.index(str(run)))
```

Comment:
note: I had to add this in the case any of the PSM run values was `None`, which OpenMS did not like. See line 215 in the reader, where I convert `"None"` back to `None`.

Reply:
Thanks for the fix! 🚀